### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by Maneren**

### CODE REVIEW

**Comment by Maneren** [comment]
I guess the comments with refactoring notes should be removed before merging.
And also remove/finish the tests, now they are just quick and dirty sketch.

File: src/lib.rs [hunk 1]
```diff
@@ -136,103 +142,152 @@ impl Dahlia {
...
 
-fn find_ansi_codes(string: &String) -> Vec<String> {
-    let patterns = [
+// a way to make a const out of (by Rust's definition) non-const code
+lazy_static! {
+    static ref CODE_REGEXES: [Regex; 2] = [
+        re(r"&(~?)([0-9a-gl-or])"),
+        re(r"&(~?)\[#([0-9a-fA-F]{6})\]"),
+    ];
+    static ref ANSII_REGEXES: [Regex; 3] = [
```
Comment (trag1c) [inline src/lib.rs]: should be ANSI, not ANSII

**Review by trag1c — CHANGES_REQUESTED** [review]
all of these are so good :yum:

ig you can remove the comments now

**Comment by Maneren** [comment]
done

**Review by trag1c — APPROVED** [review]

**Comment by trag1c** [comment]
Huge thanks :heart:

**Pull request merged**